Sources/SwiftKuery/ConnectionPoolConnection.swift

```diff
@@ -72,6 +74,24 @@ public class ConnectionPoolConnection: Connection {
         }
     }
     
+    /// Establish a connection with the database.
+    ///
+    /// - Returns: A QueryError if the connection cannot connect, otherwise nil
+    public func connectSync() -> QueryError? {
+        var error: QueryError?
+        let semaphore = DispatchSemaphore(value: 0)
+        connect() { err in
+            error = err
+            semaphore.signal()
+        }
+        semaphore.wait()
+        guard let errorUnwrapped = error else {
```

Comment:
Can't we just `return error`?

Reply:
Yes, hangover from cut and pasting other sync code. Updated